qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/java/AbstractQuteTemplateLinkCollector.java

```diff
@@ -118,7 +121,7 @@ public boolean visit(FieldDeclaration node) {
 							.getLocationExpressionFromConstructorParameter(variable.getName().getIdentifier());
 				}
 				String fieldName = variable.getName().getIdentifier();
-				collectTemplateLink(node, locationExpression, getTypeDeclaration(node), null, fieldName);
+				collectTemplateLink(node, locationExpression, getTypeDeclaration(node), null, fieldName, false);

```

Comment:
I can't seem to get the links that appear on fields to show the fact that they link to a fragment. For instance if I have the field:

```java
@Inject
@Location("myCoolTemplate.html")
Template myCoolTemplate$myFragment;
```

I get a code lens that links to the template and clicking it works as expected, but nothing is listed about the fragment in the code lens message.

Reply:
Is it allowed to write a field with $ ?

Reply:
Then `ignoreFragments` should be `true` here, right?

Reply:
According to my understand of the doc, the field name (which can contain $) has not the same behavior of method name of CheckedTemplate.

See https://quarkus.io/guides/qute-reference#fragments

```java
@Inject
Template item;

String useTheFragment() {
   return item.getFragment("item_aliases") 
            .data("aliases", List.of("Foo","Bar")) 
            .render();
}
```

For field, fragment is supported when  Template#getFragment is called. I don't support this usecase for the moment. But it makes no senses to write a Template instance field with $ to manage fragment.